### PR TITLE
Remove uses of CLRJIT_AZ_KEY/clrjit_key1 from SPMI

### PR DESCRIPTION
**Pull request by EgorBo**
Instead, we use `azure.identity` azure package and `AzureCliCredential`

### CODE REVIEW

File: eng/pipelines/common/xplat-setup.yml [hunk 1]
```diff
@@ -181,7 +181,7 @@ jobs:
         # Official Build Windows Pool
         ${{ if and(or(eq(parameters.osGroup, 'windows'), eq(parameters.jobParameters.hostedOs, 'windows')), ne(variables['System.TeamProject'], 'public')) }}:
           name: $(DncEngInternalBuildPool)
-          demands: ImageOverride -equals windows.vs2022.amd64
+          demands: ImageOverride -equals 1es-windows-2022 # ???
```
Comment (kunalspathak) [inline eng/pipelines/common/xplat-setup.yml:184]: so this needs fixing yet.

**Comment by JulieLeeMSFT** [comment]
CC @agocke @jkoritzinsky.

**Comment by JulieLeeMSFT** [comment]
@hoyosjs, we need your code review that removes the Azure Key from SPMI collection.

**Comment by jkoritzinsky** [comment]
Can we see a test run of the superpmi-collect pipeline with these changes (and a different JIT-EE guid to avoid overwriting existing collections)?

File: eng/pipelines/coreclr/templates/upload-jits-steps.yml [hunk 2]
```diff
@@ -10,10 +10,15 @@ steps:
...
-  env:
-    CLRJIT_AZ_KEY: $(clrjit_key1) # secret key stored as variable in pipeline
+- task: AzureCLI@2
+  displayName: 'Upload JIT to Azure Storage'
+  inputs:
+    azureSubscription: 'superpmi-collect-rw'
+    scriptType: 'pscore'
+    scriptLocation: 'inlineScript'
+    inlineScript: |
+      $Env:AZCOPY_AUTO_LOGIN_TYPE="AZCLI"
```
Comment (hoyosjs) [inline eng/pipelines/coreclr/templates/upload-jits-steps.yml:23]: Do you shell onto azcopy?

**Review by hoyosjs — APPROVED** [review]
From an Azure.Identity perspective, this looks good. As long as you've tested the identity has all necessary scopes and it's a federated connection, looks good for Wave 1 reqs.

**Comment by hoyosjs** [comment]
Just checked and the perms on the MI sound reasonable on the account and the service connection also looks good.

**Comment by EgorBo** [comment]
> Can we see a test run of the superpmi-collect pipeline with these changes (and a different JIT-EE guid to avoid overwriting existing collections)?

I've kicked off two runs:

- [spmi-collect](https://dev.azure.com/dnceng/internal/_build/results?buildId=2486784&view=results)
- [jitrollingbuild](https://dev.azure.com/dnceng/internal/_build/results?buildId=2486785&view=results) 

Since both of them use the key.

File: eng/pipelines/coreclr/templates/upload-jits-steps.yml (hunk 2 above)
Comment (EgorBo) [reply eng/pipelines/coreclr/templates/upload-jits-steps.yml:23]: @hoyosjs not sure I understand the question. we don't use `azcopy` here explicitly, but we use `AzureCliCredential` provider inside the python script we invoke here.
Perhaps, we can also remove it and use `DefaultAzureCredential` I didn't test it - is it an issue?

File: eng/pipelines/common/xplat-setup.yml (hunk 1 above)
Comment (EgorBo) [reply eng/pipelines/common/xplat-setup.yml]: looks like it's not - works without this change just fine, reverted

**Comment by EgorBo** [comment]
> I've kicked off two runs

Seem to be working correctly. Do we need to save our current key somewhere or it's fine to just delete it?

**Comment by kunalspathak** [comment]
> > I've kicked off two runs
> 
> Seem to be working correctly. Do we need to save our current key somewhere or it's fine to just delete it?

Just make sure we have backup. lets talk offline.

**Review by jkoritzinsky — APPROVED** [review]
Code and pipeline runs look good to me.

File: eng/pipelines/coreclr/templates/upload-jits-steps.yml (hunk 2 above)
Comment (hoyosjs) [reply eng/pipelines/coreclr/templates/upload-jits-steps.yml:23]: It's not an issue - but `DefaultAzureCredential` should be enough. The only caveat BTW is the possibility that DefaultAzureCredential might pick up other ambient identities (MI's and Workload identities) before the Az CLI creds. if you ever hit this, just swap `DefaultAzureCredential` for `AzureCliCredential`

Comment (EgorBo) [reply eng/pipelines/coreclr/templates/upload-jits-steps.yml]: Replaced `AzureCliCredential` with `DefaultAzureCredential`

**Pull request merged**